src/output.cpp

```diff
@@ -0,0 +1,736 @@
+    "                ####################     %%%%%%%%%%%%%%%%%%%%%%\n" <<
+    "                #####################     %%%%%%%%%%%%%%%%%%%%%\n" <<
+    "                ######################     %%%%%%%%%%%%%%%%%%%%\n" <<
+    "                #######################     %%%%%%%%%%%%%%%%%%\n" <<
+    "                 #######################     %%%%%%%%%%%%%%%%%\n" <<
+    "                 ######################     %%%%%%%%%%%%%%%%%\n" <<
+    "                  ####################     %%%%%%%%%%%%%%%%%\n" <<
+    "                    #################     %%%%%%%%%%%%%%%%%\n" <<
+    "                     ###############     %%%%%%%%%%%%%%%%\n" <<
+    "                       ############     %%%%%%%%%%%%%%%\n" <<
+    "                          ########     %%%%%%%%%%%%%%\n" <<
+    "                                      %%%%%%%%%%%\n\n";
+
+  // Write version information
+  std::cout <<
+    "                   | The OpenMC Monte Carlo Code\n" <<
+    "         Copyright | 2011-2019 MIT and OpenMC contributors\n" <<
+    "           License | http://openmc.readthedocs.io/en/latest/license.html\n" <<
+    "           Version | " << VERSION_MAJOR << '.' << VERSION_MINOR << '.'
+    << VERSION_RELEASE << (VERSION_DEV ? "-dev" : "") << '\n';
+#ifdef GIT_SHA1
+  std::cout << "          Git SHA1 | " << GIT_SHA1 << '\n';
+#endif
+
+  // Write the date and time
+  std::cout << "         Date/Time | " << time_stamp() << '\n';
+
+#ifdef OPENMC_MPI
+  // Write number of processors
+  std::cout << "     MPI Processes | " << mpi::n_procs << '\n';
+#endif
+
+#ifdef _OPENMP
+  // Write number of OpenMP threads
+  std::cout << "    OpenMP Threads | " << omp_get_max_threads() << '\n';
+#endif
+  std::cout << '\n';
+}
+
+//==============================================================================
+
+std::string
+header(const char* msg) {
+  // Determine how many times to repeat the '=' character.
+  int n_prefix = (63 - strlen(msg)) / 2;
+  int n_suffix = n_prefix;
+  if ((strlen(msg) % 2) == 0) ++n_suffix;
+
+  // Convert to uppercase.
+  std::string upper(msg);
+  std::transform(upper.begin(), upper.end(), upper.begin(), ::toupper);
+
+  // Add ===>  <=== markers.
+  std::stringstream out;
+  out << ' ';
+  for (int i = 0; i < n_prefix; i++) out << '=';
+  out << ">     " << upper << "     <";
+  for (int i = 0; i < n_suffix; i++) out << '=';
+
+  return out.str();
+}
+
+std::string header(const std::string& msg) {return header(msg.c_str());}
+
+void
+header(const char* msg, int level) {
+  auto out = header(msg);
+
+  // Print header based on verbosity level.
+  if (settings::verbosity >= level)
+    std::cout << '\n' << out << "\n\n";
+}
+
+//==============================================================================
+
+std::string time_stamp()
+{
+  std::stringstream ts;
+  std::time_t t = std::time(nullptr);   // get time now
+  ts << std::put_time(std::localtime(&t), "%Y-%m-%d %H:%M:%S");
```

Comment:
Optional: If you're expecting this might be used for debugging, might be useful to add millisecond precision.  Can make a difference when comparing and ordering log outputs.